pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -198,6 +198,11 @@ func validateContainerPorts(ports []corev1.ContainerPort) *apis.FieldError {
 		errs = errs.Also(apis.ErrDisallowedFields(disallowedFields...))
 	}
 
+	// Don't allow userPort to conflict with QueueProxy sidecar
```

Comment:
This is a little weird. It seems better to move around the sidecar than to block usage of that port altogether.

Short term, this seems better than causing breakage, but it would be good to avoid random blockages like this in the user experience long term. (Particularly since we may add or remove ports in the future depending on our needs.)

Reply:
@evankanderson  I agree. I think this is a good short-term fix to prevent subtlety broken configurations, but I do think we may need to explore randomly assigned ports for sidecars later.

Reply:
This SGTM but can we record it in an issue somewhere - https://github.com/knative/serving/issues/2752 seems specific to 'silently failing' so maybe a new one?

Reply:
Opened #2761